src/components/Inventory/Inventory.test.js

```diff
@@ -1,95 +1,107 @@
+
+const StubInventory = ({ gameState = {}, ...overrides }) => {
+  return (
+    <FarmhandContext.Provider
+      value={{ gameState: { ...defaultGameState, ...gameState }, handlers: {} }}
+    >
+      <Inventory
+        items={[]}
+        playerInventory={[]}
+        shopInventory={[]}
+        {...overrides}
+      />
+    </FarmhandContext.Provider>
   )
-})
+}
+describe('Inventory Component', () => {
+  describe('Displaying items', () => {
+    test('displays all items when no categories are selected', () => {
+      const items = [
+        testItem({ id: 'carrot', name: 'Carrot' }),
+        testItem({ id: 'pumpkin', name: 'Pumpkin' }),
+        testItem({ id: 'carrot-seed', name: 'Carrot Seed' }),
+      ]
+      render(<StubInventory items={items} selectedCategories={[]} />)
+      items.forEach(item => {
+        expect(screen.getByText(item.name)).toBeInTheDocument()
+      })
+    })
+
+    test('filters items by search query', () => {
+      const items = [
+        testItem({ id: 'carrot', name: 'Carrot', category: 'CROPS' }),
+        testItem({
+          id: 'pumpkin-seed',
+          name: 'Pumpkin Seed',
+          category: 'SEEDS',
+        }),
+      ]
+
+      render(<StubInventory items={items} selectedCategories={[]} />)
 
-describe('rendering items', () => {
-  test('shows the inventory', () => {
-    component.setProps({ items: [testItem({ id: carrot.id })] })
+      const searchInput = screen.getByPlaceholderText('Search inventory...')
+      fireEvent.change(searchInput, { target: { value: 'Carrot' } })
 
-    const li = component.find('li')
-    expect(li).toHaveLength(1)
-    expect(li.find(Item)).toHaveLength(1)
+      vitest.runAllTimers()
+
+      expect(screen.getByText('Carrot')).toBeInTheDocument()
+      expect(screen.queryByText('Pumpkin Seed')).not.toBeInTheDocument()
+    })
   })
-})
 
-describe('item sorting', () => {
-  test('sorts by type and base value', () => {
-    expect(
-      sortItems([
+  describe('SearchBar functionality', () => {
+    test('renders SearchBar with correct placeholder', () => {
+      render(
+        <StubInventory
+          items={[]}
+          selectedCategories={[]}
+          searchQuery=""
+          onSearch={() => {}}
+        />
+      )
+
+      const searchBar = screen.getByPlaceholderText('Search inventory...')
+      expect(searchBar).toBeInTheDocument()
+    })
+  })
+
+  describe('Item sorting and categorization', () => {
+    test('sorts items by type and base value', () => {
+      const sortedItems = sortItems([
         testItem({ id: pumpkinSeed.id, value: 0.5 }),
         testItem({ id: 'scarecrow' }),
         testItem({ id: 'sprinkler' }),
         testItem({ id: carrotSeed.id }),
       ])
-    ).toEqual([
-      testItem({ id: carrotSeed.id }),
-      testItem({ id: pumpkinSeed.id, value: 0.5 }),
-      testItem({ id: 'sprinkler' }),
-      testItem({ id: 'scarecrow' }),
-    ])
-  })
 
-  test('divides into type categories', () => {
```

Comment:
This test seems valuable. Would you mind reverting this removal (and make any necessary updates to make it pass)?